### PR TITLE
Revise intro in README for visual, logic issues

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
This PR updates the README, primarily in the intro, to address #177 and #178 .  This PR also adds a note to the technical overview picture, explaining what part of the system is doing each of the actions between clusters.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,11 +4,20 @@
...
 
-- APIs should be bindable into a cluster and operated by a service provider
-- these APIs should not require (custom) controllers/operators run locally in the consuming cluster
-- only a single vendor-neutral, OpenSource agent should be required.
+## What is it?
+
+kube-bind is a prototype project that aims to provide better support for service providers and consumers that reside in distinct Kubernetes clusters.
+
+- A service provider defines its API in terms of CRDs and associated permission claims/limitations, and exports it for use from other clusters.
+- Service consumers identify the services they want to consume.
```
Comment (pkbhowmick) [inline README.md:14]: `Consumer` itself sounds better than `Service consumer` to me. You might have different thought.

Comment (pkbhowmick) [inline README.md:14]: What about this? 
`Consumers choose the service they want to consume and sent a bind request through the service provider exposed API`

File: README.md [hunk 2]
```diff
@@ -4,11 +4,20 @@
...
+- The service CRDs get installed in the service consumer clusters, with objects of the defined kinds written and read by the service consumers.
+- The service provider indirectly reads and writes those objects as the interface to the service that it provides.
```
Comment (pkbhowmick) [inline README.md:16]:
```suggestion
- The service provider indirectly reads and writes(based on the permission given) those objects as the interface to the service that it provides.
```

**Review by pkbhowmick — CHANGES_REQUESTED** [review]
Hi @MikeSpreitzer 
thanks for the PR. I just have some thought on it. Overall looks good!

**Comment by sttts** [comment]
I like the direction. Leaving discussion about wording to @pkbhowmick. Thanks @MikeSpreitzer for the contribution!

File: README.md (hunk 2 above)
Comment (MikeSpreitzer) [reply README.md:16]: Is this even true?  I have a concern in the opposite direction: that word "indirectly" in my proposal is covering a lot of information --- the fact that there are copies of the service API objects in the service provider's cluster and the konnector is copying state back and forth between the two copies of a given service API object.

File: README.md [hunk 3]
```diff
@@ -4,11 +4,20 @@
...
+You are invited to [contribute](#contributing)!
 
-- APIs should be bindable into a cluster and operated by a service provider
-- these APIs should not require (custom) controllers/operators run locally in the consuming cluster
-- only a single vendor-neutral, OpenSource agent should be required.
+## What is it?
+
+kube-bind is a prototype project that aims to provide better support for service providers and consumers that reside in distinct Kubernetes clusters.
+
+- A service provider defines its API in terms of CRDs and associated permission claims/limitations, and exports it for use from other clusters.
```
[the remark this answers is not in the document]
Comment (MikeSpreitzer) [reply README.md:13]: I do not see permission limitations as a negative thing.  The ability to control access is something that people know they want.

File: README.md (hunk 1 above)
Comment (MikeSpreitzer) [reply README.md:14]: Is this accurate?  I thought that the service consumer creates a local API object declaring the desire to consume a service.

Comment (pkbhowmick) [reply README.md:14]: Here API means HTTP API.

Comment (pkbhowmick) [reply README.md:14]: okay, happy with it

File: README.md (hunk 2 above)
Comment (pkbhowmick) [reply README.md:16]: Consumer creates object on the consumer cluster and then konnector creates it in the Provider side. Konnector does status down-sync(provider->consumer) & spec up-sync(consumer).

Comment (pkbhowmick) [reply README.md:16]: Provider doesn't write anything in the spec on the Consumer side. So write seems bit generic to me.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (pkbhowmick) [reply README.md:13]: ah, I misunderstood earlier :D

File: README.md (hunk 2 above)
Comment (pkbhowmick) [reply README.md:16]: btw, let's merge it. much better info than before

**Review by pkbhowmick — APPROVED** [review]

**Pull request merged**